Issue/settings dialog

cayb0rg opened the pull request:
Fixes #1512 
- Hides embedded only option for students and student made widgets and increases dialog's max-height

Fixes #1504 
- Updates settings dialog view and edit permissions for grandfathered-in users

Additional
- Added error message for when you try to remove guest access from a student-made widget in instance manager
- Adds 'owner' tag to collaborator dialog

Continuation of #1516

clpetersonucf commented on src/components/my-widgets-scores.jsx:
```diff
@@ -61,7 +61,9 @@ const MyWidgetsScores = ({inst, beardMode}) => {
 	let contentRender = <LoadingIcon />
-	if (isFetched) {
+	if (is_student) {
+		contentRender = <p>Students cannot view scores.</p>
```
Could we modify this so that students can only view scores if the widget is in guest mode? Historically, we've been fine with student users being able to view score results, just anonymized ones. An even better solution would be to potentially filter out score results except for "All Guests", in edge cases where a student is given access to a widget that previously had non-anonymized scores.

clpetersonucf reviewed (changes requested):
Great work. I left one request below based on displaying scores, but I'd also like to see some level of responsiveness added to the settings dialog. Currently, if a user is interacting with the dialog with a lower resolution display, the bottom of the dialog is cut off and prevents them from selecting the Save and Cancel options at the bottom.

cayb0rg commented:
I overlooked the fact that students should still be able to view guest scores, thank you! I added an `is_student` check to `play_logs_get` to exclude information such as user ID, first name, last name, and username from the results. The front-end will combine these into a single "Guests" score. It will still display the number of students.

I've also updated the setting dialog to be more responsive.

clpetersonucf commented on src/components/my-widgets-scores.jsx:
```diff
@@ -7,7 +7,7 @@ import LoadingIcon from './loading-icon'
 import './my-widgets-scores.scss'
 
-const MyWidgetsScores = ({inst, beardMode}) => {
+const MyWidgetsScores = ({inst, beardMode, is_student}) => {
```
It looks like this prop is passed in but never used.

clpetersonucf commented on src/components/my-widgets-scores.scss:
```diff
@@ -10,6 +10,12 @@
 	overflow-y: auto;
 	padding-bottom: 1em;
 
+	&.limited_because_student {
```
I'm assuming this was related to the `is_student` prop you're passing to `MyWidgetsScores`, but the wires didn't quite get hooked up.

clpetersonucf reviewed (changes requested):
Some small things here and there that caught my eye.

Additionally, there appear to be some issues with rendering the settings dialog related to the responsive styles you added, at least on my end:

<img width="669" alt="Screenshot 2024-01-23 at 2 44 25 PM" src="https://github.com/ucfopen/Materia/assets/1268547/9c098850-2af9-4a4b-a490-a723ead2c88c">

cayb0rg commented:
Thanks for spotting these!

> Additionally, there appear to be some issues with rendering the settings dialog related to the responsive styles you added, at least on my end:
> 
> <img alt="Screenshot 2024-01-23 at 2 44 25 PM" width="669" src="https://private-user-images.githubusercontent.com/1268547/299089873-9c098850-2af9-4a4b-a490-a723ead2c88c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyMDA1NzYsIm5iZiI6MTcwNjIwMDI3NiwicGF0aCI6Ii8xMjY4NTQ3LzI5OTA4OTg3My05YzA5ODg1MC0yYWY5LTRhNGItYTQ5MC1hNzIzZWFkMmM4OGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjVUMTYzMTE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWRjZDk1ZjY4MDBkZDM0OWJlMDZlOWI1ZDllNWU0ZmNhMDYyZGMwY2IwYWQ4YTM4MDczZGViZGIwYjU1YzZiNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.A50IMe_t3Hqp5GfLJy4M4Q1onBEx4kFcEzFqSWxFwh8">

I couldn't replicate this issue (which I assume is the date picker wrapping even though there is room for it). Could you possibly specify the browser, screen width, and/or zoom level?

clpetersonucf commented:
> I couldn't replicate this issue (which I assume is the date picker wrapping even though there is room for it). Could you possibly specify the browser, screen width, and/or zoom level?

Firefox, 100% zoom, 1792px. I don't recall encountering this issue previously, but it definitely appears related to the addition of `flex-wrap` on line 301 of the settings-dialog stylesheet.

clpetersonucf reviewed (approved):
Approved. Really great work here: your solution to score display when a user becomes a student is inspired. It's non-destructive, still gives user score access, but decontextualizes the scores until/unless the student role is replaced.

Well done, I think this is good to go.
